opal/mca/hwloc/base/hwloc_base_util.c

```diff
@@ -389,12 +401,7 @@ int opal_hwloc_base_get_topology(void)
             return OPAL_ERROR;
         }
         free(val);
-        /* filter the cpus thru any default cpu set */
```

Comment:
We do need to pass thru "filter_cpus" in the case where we get the topology from other than shmem - was there some reason to remove it? I can see not doing so if you pull the topology from a file, and we obviously cannot do it if the topology is in shmem, but it is necessary otherwise.

Reply:
This is the case where it gets the topology from the xml file that is set created by PMIx, whos topology should be the same out of shmem. If there's a reason it's needed here I can add it though.

I notice that the code didn't filter cpus when loading the topology manually which is now the else clause on the bottom. I will add that.

The other case, when the user provided its own file, also didn't filter cpus.

Reply:
We do need it in this case (where the topology comes from the xml file) as the file is generally written by the daemon, which is not subject to the user's cpu specification. We don't do it in the case where the user provides the file because that is most commonly a developer debug use-case, not something we see done in practice, though one could make an argument that we should do it there too (so a developer could use one known topology and still test a variety of cgroup specifications against it).

We technically should do it even when we get the topology from shmem, but we can't due to the read-only nature of the shmem connection versus the way we chose to cache the results of the filter. This is something we'll have to address at some point (probably need a "shadow" tree that contains just the filtered information we used to store under "userdata").

Reply:
Understood.  I updated the PR to put that block of code back. I might have time to take a stab at addressing the shmem issue in the coming week.